### DISCUSSION TITLE
Form Field Error Message

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#228] [Component suggestions]
<img width="480" alt="visuele weergave van het form field component, pijl wijst naar de error message" src="https://github.com/nl-design-system/backlog/assets/545207/daba9059-20d1-41db-a378-fb67095d3cfb">

We willen graag meer leren over het Form Field Error Message component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1709819774734029)

---

## Naam
Form Field Error Message

## Doel
Toont een foutmelding die informeert wat er is fout gegaan en hoe dit is op te lossen.

## Gerelateerde componenten
[Invalid Form Alert](https://github.com/orgs/nl-design-system/discussions/278)

## Richtlijnen
[Foutmeldingen in een formulier - NL Design System](https://www.nldesignsystem.nl/richtlijnen/formulieren/foutmeldingen)
Onze richtlijnen rondom foutmeldingen in formulieren.

[Schrijfwijze van foutmeldingen - GitHub Discussion](https://github.com/orgs/nl-design-system/discussions/363)
GitHub Discussion waarbij we tot een schrijfwijze voor foutmeldingen proberen te komen.

## Onderzoek

[Avoid Messages Under Fields - Adrian Roselli](http://adrianroselli.com/2017/01/avoid-messages-under-fields.html)
Artikel beschrijft waarom Form Field Error Messages beter tussen het label en form control geplaatst kunnen worden.

[Avoid Default Field Validation - Adrian Roselli](https://adrianroselli.com/2019/02/avoid-default-field-validation.html)
Artikel beschrijft waarom de standaard HTML5 validatie beter vermeden kan worden. Jules Ernst geeft dit ook aan in zijn [comment op de backlog](https://github.com/nl-design-system/backlog/issues/122#issuecomment-846374674).

[Error Message Guidelines - Nielsen Norman Group](https://www.nngroup.com/articles/error-message-guidelines/)
Richtlijnen voor foutmeldingen waarbij enkele punten betrekking hebben tot Form Field Labels.

[The problem with live validation and what to do instead - Adam Silver](https://adamsilver.io/blog/the-problem-with-live-validation-and-what-to-do-instead/)
Artikel beschrijft waarom live validatie beter vermeden kan worden.

[How to make inline error messages accessible - Hidde de Vries](https://hidde.blog/how-to-make-inline-error-messages-accessible/)
Artikel waarin Hidde beschrijft hoe je inline error messages toegankelijk kunt aanbieden.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-forms-error-message--docs)
- [Storybook - Form Field with validation](https://designsystem.amsterdam/?path=/docs/components-forms-field--docs#with-validation)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/error-message)

---

### Screenshots / Varianten

**Storybook**

<img width="1019" height="178" alt="Scherm­afbeelding 2025-12-12 om 14 06 58" src="https://github.com/user-attachments/assets/0c8be611-5b43-43d9-8c24-20dc6a1d2382" />

Custom icon

<img width="1033" height="146" alt="Scherm­afbeelding 2025-12-12 om 14 08 43" src="https://github.com/user-attachments/assets/19118600-4dc5-4ad6-821b-2640ec78b56d" />

---

### Notities
- Wordt vooraf aan een form control geplaatst.
- Naast de error message lijken de meeste form controls een rode border te krijgen bij invalid state. En krijg form field een rode border aan de linker zijde.
- Error messages worden automatisch voorafgegaan door een visueel verborgen tekst, "Invoerfout". Deze tekst is te wijzigen.

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-form-field-error-message--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/FormFieldErrorMessage)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=1-1653&mode=design&t=pbACC6679d52NI2X-4) (in combinatie met Text input)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=851-944&mode=design&t=49LmhOkQMS39ii9N-4)

---

### Screenshots / Varianten

**Storybook**

![Scherm­afbeelding 2024-03-13 om 12 42 34](https://github.com/nl-design-system/backlog/assets/545207/45b8fdb3-22c8-4392-8e53-316c1785cc69)

In combinatie met Text input
![Scherm­afbeelding 2024-03-13 om 12 43 54](https://github.com/nl-design-system/backlog/assets/545207/e6585e0a-11c6-4d11-9045-7ab57f01a555)

**Figma - Huidig**
Niet beschilbaar als 'los' component maar wel toegepast in combinatie met verschillende form controls. Onder andere...

Text input
![Scherm­afbeelding 2024-03-13 om 12 44 33](https://github.com/nl-design-system/backlog/assets/545207/23ef3027-420d-45ef-8a18-6503296bd6c0)

Date picker
![Scherm­afbeelding 2024-03-13 om 12 44 33](https://github.com/nl-design-system/backlog/assets/545207/72cde381-4796-43e0-b25e-83bc3aac68c0)

Text area
![Scherm­afbeelding 2024-03-13 om 12 48 09](https://github.com/nl-design-system/backlog/assets/545207/bb2cb41f-23c6-409d-bcde-5deb4f36086f)

Radio group
![Scherm­afbeelding 2024-03-13 om 12 45 26](https://github.com/nl-design-system/backlog/assets/545207/0a03673e-3954-4e9a-ad5a-27cd1fcbec6c)


**Figma - WIP**
![Scherm­afbeelding 2024-03-13 om 12 43 27](https://github.com/nl-design-system/backlog/assets/545207/5f8c4624-fbba-4cf2-ab6a-b5bd8049325a)

![Scherm­afbeelding 2024-03-13 om 08 40 15](https://github.com/nl-design-system/backlog/assets/545207/88cf8284-4cec-4521-a2a7-6828f945437f)

---

### Notities
- In Storybook ook een Invalid state voor [Form field description](https://nl-design-system.github.io/denhaag/?path=/docs/css-input-form-field--docs).
- In Storybook wordt de error message onder een form control geplaatst.
- In Storybook wordt de error message onder de description geplaatst, en zit het form control er tussen.
- In Storybook naast de error message ook een dikke rode border rondom form control bij een invalid state.
- In Figma - Huidig wordt de error message onder een form control geplaatst.
- In Figma - Huidig wordt de error message combineerd met de description.
- In Figma - Huidig naast de error message ook een rode border en icoon (top-right) bij een invalid state.
- In Figma - WIP wordt de error message boven een form control geplaatst.
- In Figma - WIP wordt de error message onder de description geplaatst.
- In Figma - WIP staat er een icoon voorafgaand aan error message.
- In Figma - WIP naast de error message ook een rode dikke border rondom form control en links van gehele form field.

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

[Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-form-groups-input--input) (in combinatie met Text input)

---

### Screenshots / Varianten

**Storybook**
Niet beschikbaar als 'los' component maar wel toegepast in combinatie met verschillende form controls. Onder andere...

Text input
![Scherm­afbeelding 2024-03-13 om 13 00 44](https://github.com/nl-design-system/backlog/assets/545207/39a6b1f7-39de-4a7f-8a1c-b2c484cc5200)

Select
![Scherm­afbeelding 2024-03-13 om 13 01 39](https://github.com/nl-design-system/backlog/assets/545207/79fb9198-fb4e-44a4-92a7-9c83d6d672f4)

Date picker
![Scherm­afbeelding 2024-03-13 om 13 02 05](https://github.com/nl-design-system/backlog/assets/545207/c4c6041c-d69e-4d81-8ee6-01be09393574)

---

### Notities
- Wordt 'Errortext' genoemd.
- Error message wordt onder een form control geplaatst.
- Error message wordt boven de description geplaatst.
- Icoon voorafgaand aan error message.
- Naast de error message ook een rode border rondom form control bij een invalid state. Icoon binnen form control kleur deels mee?

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/textinput) (in combinatie met Text input)

---

### Screenshots / Varianten

**Documentatie website**
Niet beschikbaar als 'los' component maar wel toegepast in combinatie met verschillende form controls. Onder andere...

Text input
![Scherm­afbeelding 2024-03-13 om 13 06 08](https://github.com/nl-design-system/backlog/assets/545207/d77bc1d6-d128-402e-86d2-5b413d970469)

Radio group
![Scherm­afbeelding 2024-03-13 om 13 10 40](https://github.com/nl-design-system/backlog/assets/545207/9d528318-873c-44b0-b144-7ca315b78e08)

Text area
![Scherm­afbeelding 2024-03-13 om 13 11 00](https://github.com/nl-design-system/backlog/assets/545207/04231eac-4fb9-44ab-985f-a0d2bf3b297c)

---

### Notities
- Error message wordt boven een form control geplaatst.
- Error message wordt onder de description geplaatst.
- Naast de error message ook een rode dikke border rondom form control en links van gehele form field.

**Comment by jeffreylauwers** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/docs/pure-react-components-forms-textfield--docs) (in combinatie met Text input)

---

### Screenshots / Varianten

**Storybook**
Niet beschikbaar als 'los' component maar wel toegepast in combinatie met verschillende form controls. Onder andere...

Text input
![Scherm­afbeelding 2024-03-13 om 09 15 39](https://github.com/nl-design-system/backlog/assets/545207/6018bf10-ac29-44d7-a8ec-a3ec6a9bb0e2)

Select
![Scherm­afbeelding 2024-03-13 om 13 30 41](https://github.com/nl-design-system/backlog/assets/545207/4b17b427-5853-4dde-8cd5-19e2b15b67c6)

Radio group
![Scherm­afbeelding 2024-03-13 om 13 30 07](https://github.com/nl-design-system/backlog/assets/545207/ea680883-deff-4091-9592-4076ebf618cc)

---

### Notities
- Error message wordt onder een form control geplaatst.
- Error message wordt onder de description geplaatst.
- Naast de error message ook een gehele form field een licht rode background-color en links rode dikke border.

**Comment by jeffreylauwers** [comment]
## Open Gemeenten

[Patternlab - Forms and Search](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search/index.html)

---

### Screenshots / Varianten

**Patternlab**
Niet beschikbaar als 'los' component maar wel toegepast in combinatie met verschillende form controls. Onder andere...

Text input
![Scherm­afbeelding 2024-03-13 om 13 34 51](https://github.com/nl-design-system/backlog/assets/545207/8770d3d1-6312-4b46-8e19-58b78b1035e7)

Datepicker
![Scherm­afbeelding 2024-03-13 om 13 35 15](https://github.com/nl-design-system/backlog/assets/545207/fb858f24-2944-4f46-b5e2-7ff46bb47a2d)

Select
![Scherm­afbeelding 2024-03-13 om 13 34 58](https://github.com/nl-design-system/backlog/assets/545207/93903ed8-552b-475d-8c20-acdfcf9227cc)

Text area
![Scherm­afbeelding 2024-03-13 om 13 35 34](https://github.com/nl-design-system/backlog/assets/545207/5ed27cbe-f9a2-4bbd-a601-32fa4896737c)

Radio group
![Scherm­afbeelding 2024-03-13 om 13 35 53](https://github.com/nl-design-system/backlog/assets/545207/69d202ef-a17c-4685-a9b1-11bc674d5e31)

---

### Notities
- Error message wordt onder een form control geplaatst.
- Error message wordt [Martijn zoekt dit uit] description geplaatst.
- Naast de error message ook een rode dikke border links van gehele form field.

**Comment by jeffreylauwers** [comment]
## Rotterdam

- [GitHub](https://github.com/nl-design-system/rotterdam/tree/main/rotterdam-nlds-parent-wicket/rotterdam-nlds-wicket/src/main/java/nl/rotterdam/nl_design_system/wicket/components/form_field_error_message)
- [Figma](https://www.figma.com/design/iWuQsxelJzXVfCge87Og81/RODS-Components?node-id=1-3040&p=f&t=RvxrVWKdhV2i0wSA-0)

---

### Screenshots / Varianten

**Figma**
Toegepast in combinatie met verschillende form controls. Onder andere...

Text input / Select
![Scherm­afbeelding 2024-03-13 om 13 48 04](https://github.com/nl-design-system/backlog/assets/545207/a0f31a68-b5cc-44b3-885a-706eaddfd4a8)

Radio group
![Scherm­afbeelding 2024-03-13 om 13 48 16](https://github.com/nl-design-system/backlog/assets/545207/4117f1b3-55e7-4cf4-9bbe-749417dba096)

---

### Notities
- Error message wordt onder een form control geplaatst.
- Icoon voorafgaand aan error message.
- Naast de error message ook een rode dikke border boven de error message.

**Comment by jeffreylauwers** [comment]
## RVO

- [Documentatie website](https://nl-design-system.github.io/rvo/docs/components/form-feedback/)
- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-form-feedback--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-feedback)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=4231-18742&mode=design&t=bvr6w3e8IDvkEEp4-4)

---

### Screenshots / Varianten

**Storybook**

Error
![Scherm­afbeelding 2024-03-13 om 14 00 57](https://github.com/nl-design-system/backlog/assets/545207/8db54b67-b912-4e81-995b-a8d8e54f1c17)

Warning
![Scherm­afbeelding 2024-03-13 om 14 00 51](https://github.com/nl-design-system/backlog/assets/545207/bb54a8f2-6742-4c59-99eb-d7e9333d725c)

Toegepast in combinatie met verschillende form controls. Onder andere...

Text input
![Scherm­afbeelding 2024-03-13 om 14 06 20](https://github.com/nl-design-system/backlog/assets/545207/55f238ec-16ac-4ebb-b092-61e008483a05)

Select
![Scherm­afbeelding 2024-03-13 om 14 06 37](https://github.com/nl-design-system/backlog/assets/545207/512e7042-75c5-4c5c-b370-4ba55fed64e1)

Radio group
![Scherm­afbeelding 2024-03-13 om 14 06 52](https://github.com/nl-design-system/backlog/assets/545207/abfbd179-6e05-4d6c-b51f-edbf67235978)

---

### Notities
- Onder de naam: Form feedback.
- Error message wordt boven een form control geplaatst.
- Error message  wordt onder description geplaatst.
- Icoon voorafgaand aan error message.
- Naast de error message ook een rode dikke border rondom form control. Maar bij checkbox en radio group een dikke border links van gehele groep.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-form-field-error-message--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/form-field-error-message)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=851-944&mode=design&t=3HdSn5NQgTD7xJvM-4)

---

### Screenshots / Varianten

**Storybook**
![Scherm­afbeelding 2024-03-13 om 14 09 57](https://github.com/nl-design-system/backlog/assets/545207/8df11370-3cb8-4fd0-8629-c9aec9d1cee4)

In combinatie met Text input
![Scherm­afbeelding 2024-03-13 om 14 10 51](https://github.com/nl-design-system/backlog/assets/545207/f180eb32-f18c-427c-8f52-b6b5df0d072b)

**Figma**
![Scherm­afbeelding 2024-03-13 om 14 12 04](https://github.com/nl-design-system/backlog/assets/545207/356e7df5-7513-4091-859b-7d2b64066b7f)

![Scherm­afbeelding 2024-03-13 om 08 42 52](https://github.com/nl-design-system/backlog/assets/545207/bc1bc795-857b-4623-a357-89e30e1d5481)

---

### Notities
- Uitgebreide documentatie beschikbaar.
- Error message wordt boven een form control geplaatst.
- Error message wordt onder description geplaatst.
- Icoon voorafgaand aan error message.
- Naast de error message ook een rode dikke border rondom of border-bottom bij form control en links van gehele form field.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Form field error message op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=form+field+error+message)

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Form field error message op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=component%3A+form+field+error+message)

**Comment by jeffreylauwers** [comment]
Tijdens de Design System week 2024 vertelde Kim Denie dat de rode tekst met een roodfilter een stuk lastiger te zien is.

https://youtu.be/0k0C0pRFaP8?si=zXsLPJlb3cz0ZfWr&t=1050

**Comment by RenateRoke** [comment]
## Candidate selectieproces: Is de component al ingesteld voor meerdere huisstijlen?

Het kernteam heeft gekeken of Form Field Error Message van Gemeente Utrecht al geschikt is om richting Hall of Fame te gaan. Hiervoor willen we dat de component succesvol in tenminste 2 verschillende huisstijlen is ingezet.

We hebben gekeken naar de huisstijl van gemeente Den Haag, OpenStad, het basis thema en de huisstijl van NL Design System en we zijn tot de volgende conclusie gekomen:

🎉 Jazeker! Met jullie hulp hebben we genoeg informatie verzameld om een goed instelbaar component als Candidate beschikbaar te gaan maken.

**Comment by RenateRoke** [comment]
## Candidate selectieproces: Is de component al ingezet in meerdere projecten?

Omdat de Form Field Error Message van gemeente Utrecht geschikt is voor verschillende huisstijlen heeft het kernteam ook gekeken of hij al gebruikt wordt in productie, prototypes en templates. 
Deze informatie is nodig om de component een stap verder te brengen in het Candidate proces. 

We hebben hiervoor binnen community-projecten en in productie gekeken wie de component succesvol heeft ingezet en we zijn tot de volgende conclusie gekomen:

🎉 De component wordt onder andere gebruikt door:
- OpenStad gebruikt het in https://openstad.org/contact
- Gemeente Enschede gebruikt het samen met OpenStad in https://jijmaaktenschede.nl/directduidelijk/contact 
- Gemeente Purmerend gebruikt het in https://signalen-frontend-wcag.vercel.app/nl/incident
- NL Design System gebruikt het in https://nldesignsystem.nl/events/heartbeat/aanmelden/

Gebruik jij de Form Field Error Message ook in jouw project? Laat het ons dan weten, zodat we zoveel mogelijk input hebben voor een mooi Candidate component.

**Comment by RenateRoke** [comment]
## Candidate selectieproces: Is de component al meegenomen in een WCAG-EM audit?

Om de Form Field Error Message van Gemeente Utrecht als een toegankelijk Candidate component beschikbaar te maken is het handig als we weten of de Community implementaties al helemaal toegankelijk kunnen worden ingezet.

We hebben hiervoor gekeken of er een toegankelijkheidsverklaring beschikbaar was voor de projecten die in productie worden gebruikt. En:

🎉 Ja! We hebben de uitkomsten van de volgende WCAG-EM audit (1 of meerdere) kunnen vinden:
- Gemeente Purmerend gebruikt het in https://toegankelijkheidsrapport.frameless.io/rapport/purmerend-signalen-2024-12
- Gemeente Enschede gebruikt het middels OpenStad in https://jijmaaktenschede.nl/uploads/attachments/au0tubg6v9p5to9n1lp6qkmc-wcag-em-rapport-jijmaaktenschede-nl.pdf

Daarnaast zien we dat de Form Field Description invalid variant van Gemeente Utrecht wordt gebruikt als foutmelding in Open Formulieren. We vinden het belangrijk om eventuele toegankelijkheidsbevindingen van die implementatie ook mee te nemen in onze voorbereiding. Daarom hebben we een aantal toegankelijkheidsverklaringen bekeken van websites waarin de Form Field Description op die manier wordt ingezet:

- Gemeente Utrecht: Digitaal Loket Utrecht in https://toegankelijkheidsrapport.frameless.io/rapport/loket.digitaal.utrecht.nl-2024-06
- Open Formulieren technisch onderzoek bij gemeente Horst aan de Maas in https://www.horstaandemaas.nl/file/deelonderzoek-techniek-dimpact-formulieren-systeem-wcag-21-aa-pdf
- Open Formulieren content onderzoek bij gemeente Horst aan de Maas in https://www.horstaandemaas.nl/sites/horst-ad-maas/files/2023-03/WCAG%202.1%20AA%20inspectie%20-%20open-formulieren.horstaandemaas.nl%20-%201.1.pdf


### Notities voor toegankelijkheid
📝 We hebben wat dingen gevonden die in een Candidate implementatie beter kunnen. Omdat we die niet willen vergeten hebben we hier de notities gemaakt.

Contentbevindingen:
- Het contrast van de tekst van de foutmelding op de achtergrond is te laag.
- Er ontbreekt een foutmelding, of de tekst van de foutmelding is niet beschrijvend genoeg.
- Er ontbreekt een foutsuggestie, of de tekst van de foutsuggestie is niet beschrijvend genoeg.

Om deze fouten te voorkomen zullen we goede documentatie schrijven over het toegankelijk inzetten van deze component.

Techniekbevindingen:
- De foutmelding is geen live-region en wordt daarom niet automatisch voorgelezen aan hulpsoftware.

Deze bevinding nemen we mee wanneer we aan de slag gaan met het maken van deze candidate component.

**Comment by RenateRoke** [comment]
:tada: We gaan aan de slag om Form Field Error Message Candidate te maken!

Het kernteam gaat hiervoor komende tijd kijken in deze GitHub Discussion en in de Community implementaties die beschikbaar zijn.

Daarbij kijken we ook naar community componenten die als variant van een andere component fungeren als een Form Field Error Message, zoals de Form Field Description invalid variant van de gemeente Utrecht (momenteel in gebruik bij Open Formulieren) en de Alert van RVO.

Omdat een Error Message, Description en Alert ondanks hun overlap specifieke gebruiksdoeleinden hebben en daardoor dus ook specifieke (toegankelijkheids-) richtlijnen nodig hebben, willen wij deze componenten opzichzelfstaand aanbieden.

Wij ontvangen heel graag jullie feedback en wensen voor de Candidate Form Field Error Message, zodat we zoveel mogelijk input hebben om mee aan de slag te gaan!

**Comment by VincentSmedinga** [comment]
Willen we dit component generaliseren om andere betekenissen dan ‘fout’ te ondersteunen?

Ik zie dat RVO een oranje variant met ‘warning’ connotatie voert. De toegevoegde waarde naast de foutmelding is mij persoonlijk niet duidelijk, maar wellicht is er een rationale voor.

Ook herinner ik me een ontwerp van een groene ‘succes’ melding bij een Password Input, waarmee je aangeeft dat aan de eisen voor een wachtwoord voldaan wordt. (Ik kan die even niet vinden, vandaar geen link.) Deze kan ik ook begrijpen, al werkt die misschien alleen met validatie op de client. Immers, als er geen fout is kun je door naar de volgende pagina en zou je de succesmelding niet meer zien.

Samen met de rode foutmelding lijken deze drie varianten nogal op elkaar. Een korte tekst, eventueel een icoon, een kleur. De plaats in een Field is gelijk, de toegankelijke werking waarschijnlijk ook. Het lijkt overdreven om hier drie componenten voor te maken; aangenomen dat die andere twee varianten de moeite waard zijn. Het zou, naast een naamswijziging naar bijvoorbeeld Feedback Message, een kleine uitbreiding zijn, vergelijkbaar met bijv. Alert.

**Reply by Rozerinay** [reply]
@VincentSmedinga interessant punt. Visueel lijken deze varianten inderdaad sterk op elkaar: een korte tekst, soms een icoon en een kleur op dezelfde plek in het veld.

Tegelijk vervullen ze een andere rol in het formulier. Een Form Field Error Message hoort bij validatie en wordt getoond wanneer de invoer niet voldoet aan de regels na een validatiemoment, zoals bij [het verlaten van het veld of bij het verzenden van het formulier](https://nldesignsystem.nl/richtlijnen/formulieren/foutmeldingen/controleren/).

Voor doorlopende feedback tijdens het invoeren hebben we daarom een [Form Field Status](https://github.com/orgs/nl-design-system/discussions/424) als component geïntroduceerd. Denk bijvoorbeeld aan een Character Count of feedback bij wachtwoordeisen bij een Password Input. Deze informatie verandert terwijl de gebruiker typt en heeft daardoor een ander interactiepatroon dan validatie.

In die zin overlapt wat je beschrijft als een mogelijke “Feedback Message” grotendeels met wat we Form Field Status noemen. We hebben ervoor gekozen dat dit niet een variant van Form Field Error Message is, maar een apart component die gebruik zal maken van `role="status"`. Documentatie over deze 2 componenten kunnen we dan ook mooi gescheiden houden.

**Reply by VincentSmedinga** [reply]
@Rozerinay Ja, die modellering is inderdaad veel beter! Bedankt voor je uitleg. Ik zet Form Field Status op onze backlog.

**Comment by jeffreylauwers** [comment]
## Candidate voorbereidingsfase: Versimpeld of gesplitst

We hebben meerdere implementaties van Form Field Error Message bekeken en met elkaar vergeleken. Op basis daarvan hebben we besloten we dat er géén specifieke varianten nodig zijn voor dit Candidate component.

### Mis je een variant?
Laat het ons in deze Discussion weten als er uit gebruikersonderzoek naar voren is gekomen dat deze design-beslissing een hele verbetering zou zijn.

**💡 Tip**
Je kunt [varianten ook als extensie toevoegen](https://www.nldesignsystem.nl/handboek/componenten-vinden/#ik-mis-een-functionaliteit-binnen-een-candidate-of-hall-of-fame-component-wat-nu) aan een component.

**Comment by eslook** [comment]
## Candidate voorbereidingsfase: Issues bekend

We hebben gekeken bij Utrecht, Amsterdam, Den Haag, Rotterdam, Rijkshuisstijl Community, RVO en Open Formulieren naar openstaande issues en pull request gerelateerd aan de Form Field Error Message, de input uit de GitHub discussion meegenomen en issues bekend bij de community uitgevraagd. Daaruit zijn de volgende issues bekend gekomen.

We hebben besloten om deze issues wel mee te nemen, om de volgende redenen:

1. [Utrecht](https://github.com/nl-design-system/utrecht/)
    1. [Add token to support font-weight: bold for form-field-error-message](https://github.com/nl-design-system/utrecht/issues/2701) 
    Dit is nog niet geimplementeerd en nemen we mee naar Candidate.
    1. [feature request: add support to style error messages as an alert](https://github.com/nl-design-system/utrecht/issues/2551) welke voortkomt uit [feature request: align Form Field content](https://github.com/nl-design-system/utrecht/issues/2766)
        - De volgende tokens ondersteunen we al aan de hand van andere voorbereiding in Candidate: `column-gap`, `icon-size`.
        - De volgende tokens gaan we meenemen in Candidate: `icon.color`.
        - De volgende tokens gaan we niet meenemen: `border-radius`, `border-color`, `border-width`. De community heeft ze zelf nog niet geimplementeerd, en er zijn nog wisselende ideeen over hoe ze Alert eruit willen laten zien. Dit kan gemakkelijk een eigen aanvulling worden op dit component.
        - De icon ondersteunen we al aan de hand van andere voorbereiding in Candidate.
1. [Open Formulieren](https://github.com/open-formulieren)
    1. [Allow setting position of description and warning](https://github.com/open-formulieren/open-forms/issues/4601)
    Dit komt overeen met wat we zien in de GitHub discussies: de positie van de Form Field Description en de Form Field Error Message kan zowel onder als boven de Form Field geplaatst worden.

We hebben besloten om deze issues niet mee te nemen, om de volgende redenen:

1. [Utrecht](https://github.com/nl-design-system/utrecht/)
    1. [Should utrecht-form-field have a font-family token or not?](https://github.com/nl-design-system/utrecht/issues/2259)
    Deze issue kwam wel naar voren in de zoektocht naar Form Field Error Message issues, maar de specifieke wijziging voor Form Field Error Message is ondertussen al geimplementeerd en is dus niet relevant.
    1. [Form-field-description background-color token](https://github.com/nl-design-system/utrecht/issues/2258)
    Deze issues kwam naar voren als algemene check, omdat we vaak de `color` en `background-color` token samen beschikbaar stellen. Voor de Form Field Description/Error Message is dit niet van toepassing. We hebben de issue gesloten.
1. [Rotterdam](https://github.com/nl-design-system/rotterdam)
    1. [Form Field-componenten maken](https://github.com/nl-design-system/rotterdam/issues/385) 
    Deze issue kwam wel naar voren in de zoektocht naar Form Field Error Message issues, maar de genodigde aanpassing is al opgelost voor de Form Field Error Message.
1. [Rijkshuisstijl Community](https://github.com/nl-design-system/rijkshuisstijl-community/)
    1. [Componenten in Storybook met nieuwe Rijkshuisstijl opleveren](https://github.com/nl-design-system/rijkshuisstijl-community/issues/2316) 
    Deze issue heeft geen impact op de Candidate implementatie omdat het specifiek gaat over het toepassen van het nieuwe Rijkshuisstijl thema.
    1. [RHC Figma Changelog](https://github.com/nl-design-system/rijkshuisstijl-community/issues/2093) 
    Deze issue kwam wel naar voren in de zoektocht naar Form Field Error Message issues, maar is niet van toepassing, omdat dit een verzamel issue voor de Figma Changelog betreft.
1. [RVO](https://github.com/nl-design-system/rvo/)
    1. [Use existing NL Design System community React components where possible](https://github.com/nl-design-system/rvo/issues/265) 
    Deze issue kwam wel naar voren in de zoektocht naar Form Field Error Message issues, maar de specifieke wijziging voor Form Field Error Message is ondertussen al geimplementeerd en is dus niet relevant.
1. [Den Haag](https://github.com/nl-design-system/denhaag/)
    1. [Spacing between components](https://github.com/nl-design-system/denhaag/issues/1121) 
    Standaard houden we al rekening met de spacing tussen componenten, dit is niet iets waar we apart rekening mee hoeven te houden.

Uit Amsterdam zijn geen issues bekend gekomen.

Bekijk in [deze Candidate issue](https://github.com/nl-design-system/candidate/issues/916) hoe we tot dit resultaat zijn gekomen.

### Hoe kun je helpen om deze features wel in de Candidate implementatie te krijgen?

Pas de wijzigingen eerst in de Community implementatie toe en vraag andere organisaties in de community om deze ook in hun projecten te gebruiken. Documenteer jullie bevindingen en successen in de GitHub Discussion, deel deze in de Heartbeat en op Slack. Zo heeft het kernteam meer bewijs dat de feature wenselijk en haalbaar is en kan deze besluiten de functionaliteit alsnog, of in een volgende versie mee te nemen.

**Comment by jeffreylauwers** [comment]
## Candidate voorbereidingsfase: Anatomie bepaald

<img width="960" height="540" alt="visuele weergave van de anatomie van Form Field Error Message component" src="https://github.com/user-attachments/assets/b202dd90-6ee7-4bed-899a-3807d3f43469" />


De component bestaat uit:

1. Form Field Error Message
2. Icon (optioneel): voor het plaatsen van een icoon voorafgaand aan de tekst.
3. Content: voor de inhoud in de Form Field Error Message.

---

Deze schets is gemaakt in: [Figma](https://www.figma.com/design/0Y9Tbb373u6QGxGT6mqUnm/NL-Design-System---Doc-website---Afbeeldingen?node-id=1777-797)

**Comment by jeffreylauwers** [comment]
## Candidate voorbereidingsfase: Design tokens bepaald

De design tokens voor de Candidate component zijn bepaald. Dat hebben we gedaan op basis van alle designs die we in deze Discussion hebben gezien. Als een beslissing vaker dan 2 keer voorkomt, is het een algemeen nuttige beslissing.

### Deze design tokens nemen we mee:

**General**
- `nl.form-field-error-message.color`
- `nl.form-field-error-message.column-gap`
- `nl.form-field-error-message.font-family`
- `nl.form-field-error-message.font-size`
- `nl.form-field-error-message.font-weight`
- `nl.form-field-error-message.line-height`
- `nl.form-field-error-message.margin-block-end`
- `nl.form-field-error-message.margin-block-start`
- `nl.form-field-error-message.icon.color`
- `nl.form-field-error-message.icon.size`

### Mis je een design token?

Laat het in deze Discussion weten als er uit gebruikersonderzoek naar voren is gekomen dat deze design-beslissing een hele verbetering zou zijn.

**💡 Tip**
Je kunt design tokens ook als extensie toevoegen aan de component.

- [Voor designers: Zelf componenten uitbreiden](https://www.nldesignsystem.nl/handboek/designer/zelf-componenten-uitbreiden/)
- Voor developers: Op het moment van schrijven is de documentatie hierover nog in ontwikkeling. Vraag ons er gerust naar.
